src/screens/TapDetailsScreen.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import type { Tap, TapMutator } from 'brewskey.js-api';
+import type DAOEntityStore from '../stores/DAOEntityStore';
+
+import * as React from 'react';
+import { inject, observer } from 'mobx-react';
+import { Text, View } from 'react-native';
+import flatNavigationParams from '../common/flatNavigationParams';
+
+type Props = {|
+  id: string,
+  navigation: Object,
+  tapStore: DAOEntityStore<Tap, TapMutator>,
+|};
+
+@flatNavigationParams
+@inject('tapStore')
+@observer
+class TapDetailsScreen extends React.Component<Props> {
+  // todo find types for navigationOptions
+  static navigationOptions = ({ navigation }: Object): Object => ({
+    title: navigation.state.params.tap && navigation.state.params.tap.name,
+  });
+
+  componentDidMount() {
+    // todo with this solution title on header appears after some lag :/
+    const { tapStore, id, navigation } = this.props;
+    navigation.setParams({ tap: tapStore.getByID(id) });
```

Comment:
This seems weird that this is set here... There should be a store that handles app state that sets this when you navigate to the tap details screen. Even better would be to handle this in a store that listens to route changes.

I guess another question I have is -- why are you setting an object in `navigation`? I would think that navigation just keeps track of IDs

Reply:
1) Header component is handled by react-navigation.
2) we setup header title in static navigationOptions of screen.
3) setParams() is the way how we can pass component props to static navigationOptions.
4) so..we can use setParams and pass tap on componentDidMount to get the title or
5) we can full tap instead id as params, when navigate from another screen.
6) or don't use react-navigation header component at all.

Reply:
and making something like RouterStore seems a good idea, but this means that we will reimplement part of router logic by ourself.

Reply:
I'll make it `5` for now.

Reply:
Ok, well if you keep working on components and it seems to make sense to move this to a store that just listens for actions, I'd be fine with it.